Move form help text into legend

selfthinker opened the pull request:
This is an alternative solution to #485.

## What problem does the pull request solve?

Most groups of checkboxes and radio buttons within Elements have an additional introductory help text which isn't a hint. It's usually there to help less technical people understand that you can select multiple checkboxes, but can also be used for other purposes.

This moves that text into the legend to also make it accessible to assistive technologies like screen readers. The styling is nearly the same as before and adds a class to the current paragraph declaration to re-use its styling.

There is a bit more spacing above the help text in the new version (probably `0.26316em` which translates to 5px with default settings), even though it uses the same CSS. That is because previously the margins were collapsing as the `p` came directly after the `h3` and now they aren't anymore because the text has been moved into the legend. I'm not sure if it's worth fixing as spacing within Elements is quite inconsistent and it doesn't look wrong.
Let me know if you think I should do something about it.

## How has this been tested?

I've tested in multiple browsers via BrowserStack.

## Screenshots

### Before
<img width="549" alt="form-help-text-before" src="https://cloud.githubusercontent.com/assets/108893/26735965/1bf342ee-47bc-11e7-9427-6eac502abc35.png">

### After
<img width="551" alt="form-help-text-after" src="https://cloud.githubusercontent.com/assets/108893/26735977/27b7219a-47bc-11e7-927f-ff91cba935dd.png">

### Differences
<img width="549" alt="form-help-text-differences" src="https://cloud.githubusercontent.com/assets/108893/26736963/1e82d8b8-47c0-11e7-85ee-8d4cb081a622.png">

## What type of change is it?
- Bug fix (non-breaking change which fixes an issue)

selfthinker commented on assets/sass/elements/_forms.scss:
```diff
@@ -20,6 +20,11 @@ fieldset {
   width: 100%;
 }
 
+// Hack to let legends or elements within legends have margins in webkit browsers
```
This is quite hacky but is the least hacky solution I could find for fixing an issue in all webkits which will not allow a margin on the form hint text. I suspect this might be too controversial (a similar solution was tried in #76 by @timpaul and rejected). That's why I created an alternative solution to fix the same problem with #485.

igloosi replied:
could use  `overflow: hidden;` instead

selfthinker replied:
Ah, yes that works as well. I could have sworn I tried it but apparently not. I think that makes more sense, so will change it.

selfthinker commented:
I have added some documentation (as suggested by @gemmaleigh on #485), changed the "hack" to use `overflow:hidden` instead (as suggested by @igloosi) and re-tested in a couple of browsers.

gemmaleigh commented:
This looks good, thanks @selfthinker and @igloosi 👍
